acvm/src/lib.rs

```diff
@@ -159,8 +159,8 @@ pub trait ProofSystemCompiler {
     /// if the language and proof system does not line up.
     fn np_language(&self) -> Language;
 
-    // Returns true if the backend supports the selected black box function
-    fn black_box_function_supported(&self, opcode: &BlackBoxFunc) -> bool;
```

Comment:
I think we may want to retain something which allows a frontend to see which black box functions a backend supports without having to construct a full opcode. This is relevant for #269.

Alternatively we need to formalise some way to create dummy opcodes to test compatibility: `Opcode::BlackBoxFuncCall(BlackBoxFuncCall::dummy(BlackBoxFunc))` 

This is also needed for other opcodes where we want to check support where we don't have a concrete opcode yet. https://github.com/noir-lang/noir/pull/1282/files#diff-4b6abdc6086c59b7255bb6d107f5bc0cec0c68b18a26b0eaebad9671ee5b570fR139